binder/src/main/java/io/grpc/binder/SecurityPolicies.java

```diff
@@ -97,6 +97,27 @@ public static SecurityPolicy hasSignature(
         packageManager, packageName, ImmutableList.of(requiredSignature));
   }
 
+  /**
+   * Creates a {@link SecurityPolicy} which checks if the given package is signed with the same
+   * certificate that the current Android platform was signed with.
+   *
+   * @param packageName the package name of the allowed package.
+   * @throws NullPointerException if any of the inputs are {@code null}.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/11238")
+  public static SecurityPolicy hasSameSignatureAsPlatform(
+      PackageManager packageManager, String packageName) {
+    PackageInfo platformPackageInfo;
+    try {
+      platformPackageInfo =
+          packageManager.getPackageInfo("android", PackageManager.GET_SIGNING_CERTIFICATES);
+    } catch (PackageManager.NameNotFoundException e) {
+      throw new AssertionError(e);  // impossible; the platform package is always available.
+    }
+    Signature platformSignature = platformPackageInfo.signatures[0];
```

Comment:
In general, it's [guaranteed to have at least one element](https://developer.android.com/reference/android/content/pm/PackageInfo#signatures).

I **think** the other signatures come from key rotation of the same app. But since the "android" package doesn't refer to a regular, updatable app, I suspect it never goes through updates or key rotations (though I can't confirm it).

Still, seems like checking previous certificates doesn't hurt. Updated.